src/python/pants/backend/python/rules/python_create_binary.py

```diff
@@ -1,74 +1,38 @@
 # Copyright 2019 Pants project contributors (see CONTRIBUTORS.md).
 # Licensed under the Apache License, Version 2.0 (see LICENSE).
 
-from pants.backend.python.rules.inject_init import InjectedInitDigest
-from pants.backend.python.rules.pex import (
-  CreatePex,
-  Pex,
-  PexInterpreterConstraints,
-  PexRequirements,
-)
-from pants.backend.python.subsystems.python_setup import PythonSetup
+from pants.backend.python.rules.pex import CreatePexFromTargetClosure, Pex
 from pants.backend.python.targets.python_binary import PythonBinary
-from pants.engine.fs import Digest, DirectoriesToMerge
-from pants.engine.legacy.graph import BuildFileAddresses, HydratedTarget, TransitiveHydratedTargets
+from pants.build_graph.address import Address
+from pants.engine.legacy.graph import BuildFileAddresses, HydratedTarget
 from pants.engine.legacy.structs import PythonBinaryAdaptor
 from pants.engine.rules import UnionRule, rule
-from pants.engine.selectors import Get, MultiGet
+from pants.engine.selectors import Get
 from pants.rules.core.binary import BinaryTarget, CreatedBinary
 from pants.rules.core.strip_source_root import SourceRootStrippedSources
 
 
 @rule
-async def create_python_binary(python_binary_adaptor: PythonBinaryAdaptor,
-  python_setup: PythonSetup) -> CreatedBinary:
-  transitive_hydrated_targets = await Get(
-    TransitiveHydratedTargets, BuildFileAddresses((python_binary_adaptor.address,))
-  )
-  all_targets = transitive_hydrated_targets.closure
-  all_target_adaptors = [t.adaptor for t in all_targets]
-
-
-  interpreter_constraints = PexInterpreterConstraints.create_from_adaptors(
-    adaptors=tuple(all_targets),
-    python_setup=python_setup
-  )
-
-  source_root_stripped_sources = await MultiGet(
-    Get(SourceRootStrippedSources, HydratedTarget, target_adaptor)
-    for target_adaptor in all_targets
-  )
-
+async def create_python_binary(python_binary_adaptor: PythonBinaryAdaptor) -> CreatedBinary:
   #TODO(#8420) This way of calculating the entry point works but is a bit hackish.
   entry_point = None
   if hasattr(python_binary_adaptor, 'entry_point'):
     entry_point = python_binary_adaptor.entry_point
   else:
     sources_snapshot = python_binary_adaptor.sources.snapshot
     if len(sources_snapshot.files) == 1:
-      target = transitive_hydrated_targets.roots[0]
+      target = await Get(HydratedTarget, Address, python_binary_adaptor.address)
```

Comment:
NB: the new syntax is `await Get[HydratedTarget](Address, python_binary_adaptor.address)`. We haven't finished updating every rule yet, though, and I know you're just moving around the code so it's okay to use the old style here.